### DISCUSSION TITLE
代码运行过慢问题

### DISCUSSION THREAD

**Discussion by sjdl233** [datawhalechina/learn-python-the-smart-way-v2#27] [提问答疑]
<img width="568" alt="屏幕截图 2023-10-24 165803" src="https://github.com/datawhalechina/learn-python-the-smart-way-v2/assets/148856649/d17f8039-e629-405d-8b1c-3d51c54313b2">

这段代码在n=5以上运行速度很慢，我尝试过将hw代码用字符串方式撰写，速度会快，请问这是为什么

**Comment by anine09** [comment]
同学，我建议你先用markdown格式吧缩进体现出来先，我们无法阅读你的代码
________________________________
From: sjdl233 ***@***.***>
Sent: Tuesday, October 24, 2023 4:53:57 PM
To: datawhalechina/learn-python-the-smart-way-v2 ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [datawhalechina/learn-python-the-smart-way-v2] 代码运行过慢问题 (Discussion #27)


import ast
n = ast.literal_eval(input())
def hw(n): #回文数
a=0
while n>a:
a=a*10+n%10
n//=10
if n==a or n==a//10:
return True
else:
return False
def zs(num): #质数
if num < 2:
return False
for i in range(2, int(num ** 0.5) + 1):
if num % i == 0:
return False
return True
def hz(n):
i=0
c=-1
while True:
i=i + 1
if i>11 and i<101:
i=101
if hw(i) and zs(i):
c=c + 1
if c==n:
return i
print(hz(n))
这段代码在n=5以上运行速度很慢，我尝试过将hw代码用字符串方式撰写，速度会快，请问这是为什么

―

**Reply by sjdl233** [reply]
<img width="371" alt="屏幕截图 2023-10-24 170727" src="https://github.com/datawhalechina/learn-python-the-smart-way-v2/assets/148856649/0fc0e6c1-502f-4d73-995a-11576eead1f7">

**Reply by anine09** [reply]
你的字符串格式撰写是什么意思呢？麻烦详细描述一下。

**Reply by sjdl233** [reply]
return str(n)[::-1] == str(n)

**Reply by sjdl233** [reply]
哦哦，我发现了，是我if缩进缩多了，抱歉抱歉，打扰了
